### ISSUE TITLE
Add additional build on Switchroot Ubuntu Jammy 22.04

### ISSUE THREAD

**Issue by theofficialgman** [moonlight-stream/moonlight-qt#1208]
**Is your feature request related to a problem? Please describe.**
Currently, the bionic based build works on newer distros like 22.04+ only if libssl1.1 is pulled from an older distro (since it is not maintained and not available in newer distros).

**Describe the solution you'd like**
I would like if moonlight-qt could be built on Switchroot Ubuntu Jammy in addition to the bionic build (and for that to be added to the moonlight-qt apt repo).

**Additional context**
I am the distro maintainer for switchroot ubuntu jammy. I am happy to help setup CI or anything you might need to accomplish that.

**Comment by theofficialgman** [comment]
I also have a patchset of the l4t tegra patches on FFmpeg 6.1.1 incase that is of interest https://github.com/theofficialgman/FFmpeg/tree/6.1.1-nvv4l2
idk what you are currently using in the tegra builds

**Comment by cgutman** [comment]
Awesome to see a L4T Jammy distro.

How much longer do you all plan to maintain the L4T Bionic distro? I ask because it's a bit of a pain to support on our side because it's stuck with the oldest version of Qt of all currently supported platforms (5.9 from early 2017). If we could drop Bionic support, we could jump all the way up to Qt 5.11 (or 5.12 if we dropped Buster).

> I also have a patchset of the l4t tegra patches on FFmpeg 6.1.1 incase that is of interest

Yes it definitely is. Our L4T packages are still based on FFmpeg 4.4 IIRC, and the build scripts have some special logic to avoid using FFmpeg options that weren't supported back on 4.4. Bumping up to 6.1 like the rest of the platforms would be great.

> I am happy to help setup CI or anything you might need to accomplish that.

That would be great. The way the packages are built today is using the Dockerfiles and scripts in [my packaging repo](https://github.com/cgutman/moonlight-packaging). In particular the Dockerfile for L4T is [here](https://github.com/cgutman/moonlight-packaging/blob/master/Dockerfile.l4t.bionic).

I can easily add a Jammy version, but I need to know what base image to use for the container and what repo I should be pulling `nvidia-l4t-multimedia-utils` and dependent packages from. I believe I tried to create a L4T Focal package using Nvidia's official L4T container images before, but I encountered issues with some changes they made to the multimedia packages in JetPack 5 that broke the build.

If you could help figure out how to get the build working, I would gladly build and publish a L4T Jammy package.

**Comment by cgutman** [comment]
I started working on the Jammy dockerfile using your repo, but I ran into an issue during the FFmpeg build. Which package provides `libnvll.so`?
```
86.59 require_pkg_config libdrm libdrm xf86drm.h drmGetVersion
86.59 check_pkg_config libdrm libdrm xf86drm.h drmGetVersion
86.59 test_pkg_config libdrm libdrm xf86drm.h drmGetVersion
86.59 pkg-config --exists --print-errors libdrm
86.59 check_func_headers xf86drm.h drmGetVersion -I/usr/include/libdrm -ldrm
86.59 test_ld cc -I/usr/include/libdrm -ldrm
86.59 test_cc -I/usr/include/libdrm
86.59 BEGIN /tmp/ffconf.JwAxcKB2/test.c
86.59     1	#include <xf86drm.h>
86.59     2	#include <stdint.h>
86.59     3	long check_drmGetVersion(void) { return (long) drmGetVersion; }
86.59     4	int main(void) { int ret = 0;
86.59     5	 ret |= ((intptr_t)check_drmGetVersion) & 0xFFFF;
86.59     6	return ret; }
86.59 END /tmp/ffconf.JwAxcKB2/test.c
86.59 gcc -D_ISOC99_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_POSIX_C_SOURCE=200112 -D_XOPEN_SOURCE=600 -DPIC -I/usr/include/libdrm -std=c11 -fomit-frame-pointer -fPIC -pthread -I/usr/include/libdrm -c -o /tmp/ffconf.JwAxcKB2/test.o /tmp/ffconf.JwAxcKB2/test.c
86.59 gcc -Wl,--as-needed -Wl,-z,noexecstack -I/usr/include/libdrm -o /tmp/ffconf.JwAxcKB2/test /tmp/ffconf.JwAxcKB2/test.o -ldrm
86.59 /usr/bin/ld: warning: libnvll.so, needed by /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so, not found (try using -rpath or -rpath-link)
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllCreateImage'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllGetHdrSupported'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllGetNumHeads'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllGetNumWindows'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllGetFlipSyncpt'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllSetFlipTimestamp'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllReleaseOverlays'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllGetMode'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllDestroyImage'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllSetHead'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllCreateDevice'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllSetMode'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllDestroyDisplay'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllDestroyDevice'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllGetConnectorType'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllUnmapImage'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllVblankSyncptWait'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllImportDmaBuf'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllPflipSyncptWait'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllMapImage'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllGetEncoderType'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllIsDisplayConnected'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllSetFlipSyncpt'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllCreateImageGem'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllReadVblankSyncpt'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllSetHeadLut'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllGetHeadLut'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllSetCursor'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllMoveCursor'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllExportDmaBuf'
86.59 /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/11/../../../aarch64-linux-gnu/libdrm.so: undefined reference to `nvllGetModeDB'
86.59 collect2: error: ld returned 1 exit status
86.59 ERROR: libdrm not found using pkg-config
```

**Comment by theofficialgman** [comment]
Ah yes you are asking the wrong question but I know the answer. The error you are running into is because of one of the nvidia- packages taking control of libdrm with their own outdated and not fully functional version. I decided when making the distro to leave the nvidia- packages unmodified and correct stuff like this through other packages. 

Installing `nvidia-bsp-32-3` handles that. You will probably need to run 
`touch "/opt/nvidia/l4t-packages/.nv-l4t-disable-boot-fw-update-in-preinstall"` before installing that package since it's a meta package that installs most of the BSP.

Alternatively just do the fix yourself that I applied as part of the preinst for `nvidia-bsp-32-3` https://github.com/theofficialgman/l4t-debs-source/blob/master/deb_package_sources/main/nvidia-bsp-32-3/DEBIAN/preinst

**Comment by theofficialgman** [comment]
also I would appreciate if you could do the same thing for ubuntu `noble`. it hasn't released yet but feature freeze has already happened at ubuntu and my package repo is already in place for it. Building for it should be identical as jammy just with the dist changed.

Before you ask, I do intend to maintain both the jammy and noble releases for the foreseeable future.

**Comment by cgutman** [comment]
I've implemented builds for L4T Jammy and Noble, but I don't have any way to test them.

If you could confirm these attached locally built packages work (or build your own using my packaging scripts/containers), I can get those packaged for the next release.
[l4t-jammy-noble-builds.zip](https://github.com/moonlight-stream/moonlight-qt/files/14894857/l4t-jammy-noble-builds.zip)

**Comment by theofficialgman** [comment]
@cgutman I tested both the packages. They work perfectly. Thanks so much!

**Comment by cgutman** [comment]
Released in v6.0.0
